api/edge_api/identities/serializers.py

```diff
@@ -0,0 +1,127 @@
+        return MultivariateFeatureOption.objects.get(id=data)
+
+    def to_representation(self, obj):
+        return obj.id
+
+
+class EdgeMultivariateFeatureStateValueSerializer(serializers.ModelSerializer):
+    multivariate_feature_option = EdgeMultivariateFeatureOptionField()
+
+    def to_internal_value(self, data):
+        data = super().to_internal_value(data)
+        return EngineMultivariateFeatureStateValueModel(**data)
+
+    class Meta:
+        model = MultivariateFeatureStateValue
+        fields = (
+            "multivariate_feature_option",
+            "percentage_allocation",
+        )
+
+
+class FeatureStateValueEdgeIdentityField(serializers.Field):
+    def to_representation(self, obj):
+        identity_id = self.parent.get_identity_uuid()
+        return obj.get_value(identity_id=identity_id)
+
+    def get_attribute(self, instance):
+        # We pass the object instance onto `to_representation`,
+        # not just the field attribute.
+        return instance
+
+    def to_internal_value(self, data):
+        fsv_type = FeatureState.get_featue_state_value_type(data)
+        feature_state_value_dict = {
+            "type": fsv_type,
+            FeatureState.get_feature_state_key_name(fsv_type): data,
+        }
+        fs_value_serializer = FeatureStateValueSerializer(data=feature_state_value_dict)
+        fs_value_serializer.is_valid(raise_exception=True)
+        return FeatureStateValue(**feature_state_value_dict).value
+
+
+class EdgeFeatureField(serializers.Field):
+    def to_representation(self, obj):
+        return obj.id
+
+    def to_internal_value(self, data):
+        feature = Feature.objects.get(id=data)
+        return feature
+
+
+class EdgeIdentityFeatureStateSerializer(serializers.Serializer):
+    feature_state_value = FeatureStateValueEdgeIdentityField()
+    feature = EdgeFeatureField()
+    multivariate_feature_state_values = EdgeMultivariateFeatureStateValueSerializer(
+        many=True, required=False
+    )
+    enabled = serializers.BooleanField(required=False, default=False)
+    identity_uuid = serializers.SerializerMethodField()
+
+    featurestate_uuid = serializers.CharField(required=False, read_only=True)
+
+    def get_identity_uuid(self, obj=None):
+        return self.context["view"].kwargs["edge_identity_identity_uuid"]
+
+    def save(self, **kwargs):
+        identity = self.context["view"].identity
+        feature_state_value = self.validated_data.pop("feature_state_value")
+
+        if self.instance:
+            if self.validated_data.get("multivariate_feature_state_values"):
+                engine_multi_fs_value_models = engine_multi_fs_value_schema.load(
+                    self.validated_data["multivariate_feature_state_values"], many=True
+                )
+                self.instance.multivariate_feature_state_values = (
+                    engine_multi_fs_value_models
+                )
+            self.instance.set_value(feature_state_value)
```

Comment:
I think this line can be removed now? 

I think you can probably simplify some of this logic by just starting off with `if not instance:` and then handling the rest from there. I don't think you'd need the nested `if` statement then.